src/pkg/bigbang/bigbang.go

```diff
@@ -0,0 +1,502 @@
+	// manifests created with the provided Helm
+	template, err := zarfHelmInstance.TemplateChart()
+	if err != nil {
+		return component, fmt.Errorf("unable to template BigBang Chart: %w", err)
+	}
+
+	subPackageURLS := findURLs(template)
+	repos[0] = fmt.Sprintf("%s@%s", repos[0], component.BigBang.Version)
+	repos = append(repos, subPackageURLS...)
+
+	// Save the list of repos to be pulled down by Zarf
+	component.Repos = repos
+
+	// Get all the images.  This might be omitted here once https://github.com/defenseunicorns/zarf/issues/337
+	// is implemented
+
+	// seed the list of repos --> images
+	err = importBigbangImageList(component.BigBang.Version)
+	if err != nil {
+		return component, fmt.Errorf("unable to import bigbang image list: %w", err)
+	}
+
+	// just select the images needed to suppor the repos this configuration of BigBang will need
+	images, err := GetImages(repos)
+	if err != nil {
+		return component, fmt.Errorf("unable to get bb images: %w", err)
+	}
+
+	// dedupe the list o fimages
+	uniqueList := utils.Unique(images)
+
+	// add the images to the component for Zarf to download
+	component.Images = append(component.Images, uniqueList...)
+
+	//Create the flux wrapper around BigBang for deployment
+	manifest, err := GetBigBangManifests(componentPath.Manifests, component)
+	if err != nil {
+		return component, err
+	}
+
+	component.Manifests = []types.ZarfManifest{manifest}
+	return component, nil
+}
+
+// findURLs takes a list of yaml objects (as a string) and
+// parses it for GitRepository objects that it then parses
+// to return the list of git repos and tags needed.
+func findURLs(t string) []string {
+
+	// Break the template into separate resources
+	urls := make([]string, 0)
+	yamls, _ := utils.SplitYAML([]byte(t))
+
+	for _, y := range yamls {
+		// see if its a GitRepository
+		if y.GetKind() == "GitRepository" {
+			url := y.Object["spec"].(map[string]interface{})["url"].(string)
+			tag := y.Object["spec"].(map[string]interface{})["ref"].(map[string]interface{})["tag"].(string)
+			urls = append(urls, fmt.Sprintf("%v@%v", url, tag))
+		}
+	}
+
+	return urls
+}
+
+// GetImages identifies the list of images needed for the list
+// of repos provided.
+func GetImages(repos []string) ([]string, error) {
+
+	images := make([]string, 0)
+	for _, r := range repos {
+		// Don't try to get images for the bigbang repo
+		parts := strings.Split(r, "@")
+		if len(parts) != 2 {
+			continue
+		}
+
+		url := parts[0]
+		version := parts[1]
+
+		parts = strings.Split(url, "/")
+		packageRepo := parts[len(parts)-1]
+		packageRepo = strings.TrimSuffix(packageRepo, ".git")
+
+		images = append(images, BigBangImages[packageRepo][version]...)
+
+	}
+
+	return images, nil
+}
+
+// importBigBangFluxImageList populates an internal map of the images
+// needed for the version of flux that is defined within the provided version
+// of BigBang
+func importBigBangFluxImageList(version string) error {
```

Comment:
Instead of calling the gitlab API, can we instead pull down `https://repo1.dso.mil/big-bang/bigbang/-/raw/1.52.0/base/flux/kustomization.yaml` and then unmarshall the images out of 

```yaml
# update flux components to use ironbank images
images:
- name: ghcr.io/fluxcd/helm-controller
  newName: registry1.dso.mil/ironbank/fluxcd/helm-controller
  newTag: v0.28.1
- name: ghcr.io/fluxcd/kustomize-controller
  newName: registry1.dso.mil/ironbank/fluxcd/kustomize-controller
  newTag: v0.32.0
- name: ghcr.io/fluxcd/notification-controller
  newName: registry1.dso.mil/ironbank/fluxcd/notification-controller
  newTag: v0.30.2
- name: ghcr.io/fluxcd/source-controller
  newName: registry1.dso.mil/ironbank/fluxcd/source-controller
  newTag: v0.33.0
```


Alternatively, we could use find images because it's reliable for the flux stuff.